erpnext/hr/doctype/leave_application/leave_application.py

```diff
@@ -77,6 +77,10 @@ def validate_applicable_after(self):
 						frappe.throw(_("{0} applicable after {1} working days").format(self.leave_type, leave_type.applicable_after))
 
 	def validate_dates(self):
+		if self.from_date and self.from_date < frappe.utils.today():
+			if "HR Manager" not in frappe.get_roles():
```

Comment:
Role should  not be hard-coded, anyone can rename it. Also it should optional.

Instead, add a field "Role Allowed to Create Back-dated Leave Application" in HR Settings and validate based on it.